pkg/cmd/env.go

```diff
@@ -175,91 +136,40 @@ func DeleteEnv(ctx context.Context, args []string, ioStreams cmdutil.IOStreams)
 	if len(args) < 1 {
 		return fmt.Errorf("you must specify env name for vela env:delete command")
 	}
-	envname := args[0]
-	curEnv, err := GetCurrentEnvName()
-	if err != nil {
-		return err
-	}
-	if envname == curEnv {
-		return fmt.Errorf("you can't delete current using env %s", curEnv)
-	}
-	envdir, err := system.GetEnvDir()
-	if err != nil {
-		return err
-	}
-	if err = os.RemoveAll(filepath.Join(envdir, envname)); err != nil {
-		return err
+	envName := args[0]
+	msg, err := oam.DeleteEnv(envName)
+	if err == nil {
+		ioStreams.Info(msg)
 	}
-	ioStreams.Info(envname + " deleted")
-	return nil
+	return err
 }
 
 func CreateOrUpdateEnv(ctx context.Context, c client.Client, envArgs *types.EnvMeta, args []string, ioStreams cmdutil.IOStreams) error {
 	if len(args) < 1 {
 		return fmt.Errorf("you must specify env name for vela env:init command")
 	}
-	envname := args[0]
-	envArgs.Name = envname
-	data, err := json.Marshal(envArgs)
-	if err != nil {
-		return err
+	envName := args[0]
+	namespace := envArgs.Namespace
+	err, msg := oam.CreateOrUpdateEnv(ctx, c, envName, namespace)
+	if err == nil {
```

Comment:
```suggestion
	if err != nil {
```

go里面都是习惯了 err 不等于 nil，然后返回错误，下面再按正常逻辑走，如果你这里反过来，后面别的人在下面习惯性插入代码，就会出问题

Reply:
如果这里先写 ！= ，就需要 return 两次 err，不管 err 等于什么东西。
Golang 为什么为了遵循这个风格，显得非常重复？快把人逼疯了。感觉整个业务逻辑大多数时间都在做错误处理：）
![image](https://user-images.githubusercontent.com/2805315/90493277-c2455100-e174-11ea-96ac-4268b97f8eb2.png)

Reply:
当然，我这里有种 try catch finally 的感觉，我想先把核心的信息处理了，就是打印这个核心信息 https://github.com/cloud-native-application/RudrX/blob/master/pkg/cmd/env.go#L227，不然，我得这么写，最后一句`return nil`也是为了生生凑函数的返回类型，而其实它就是 err 的内容，不需要额外构造的：）
```
	if err != nil {
		return err
	}
	ioStreams.Info(msg)
	return nil
```

Reply:
我还是应该遵循这个规范，大家都是这么写的：）

Reply:
go 就是这个风格，follow吧，好处就是非常清晰